root/DBTaxProject/src/main/java/org/dbpedia/dbtax/database/EdgeDB.java

```diff
@@ -0,0 +1,116 @@
+package org.dbpedia.dbtax.database;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+
+
+public class EdgeDB {
+
+	/*
+	 * This function is to insert an Edge which captures the 
+	 * Parent-Child Relationship among categories
+	 */
+	public static void insertEdge(int parentId, int chidId){
+
+		// Establish Database Connection	
+		Connection connection = DatabaseConnection.getConnection();
+				
+		PreparedStatement ps = null;
+		String query = "INSERT IGNORE INTO edges(parent_id,child_id) VALUES (?, ?)";
+
+		try{
+			ps = connection.prepareStatement(query);
+			ps.setInt(1, parentId);
+			ps.setInt(2, chidId);
+			ps.executeUpdate();
+			connection.close();
+		}catch(SQLException e){
+			e.printStackTrace();
+		}
+	}
+	
+	/*
+	 * This method is used to get all parents of the given leaf Node
+	 */
+	public static ArrayList<Integer> getTransitiveParents(int leafNode){
+		
+		//Connect to database
+		Connection connection = DatabaseConnection.getConnection();
+		
+		PreparedStatement ps = null;
+		String query =  "SELECT parent_id FROM edges WHERE child_id =?";
+		
+		ResultSet rs = null;
+		
+		try{
```

Comment:
it is easier to use the try with resources feature introduced in Java 7+
http://tutorials.jenkov.com/java-exception-handling/try-with-resources.html